### ISSUE TITLE
ansible_user_* facts inconsistent with defaults across sudo/su become_method

### ISSUE THREAD

**Issue by ed-velez** [ansible/ansible#70512]
##### SUMMARY
Currently the `ansible_user_*` facts are inconsistent based on the underlying `become_method` used when they were collected.

The `ansible_user_*` facts when running in an escalated root context and `su` is used as the `become_method` will reflect the `ansible_user`:
```
host1.example.com | SUCCESS => {
    "ansible_facts": {
        "ansible_user_dir": "/home/user1",
        "ansible_user_gecos": "",
        "ansible_user_gid": 1000,
        "ansible_user_id": "user1",
        "ansible_user_shell": "/bin/bash",
        "ansible_user_uid": 1000,
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false
}
``` 

The `ansible_user_*` facts when running in an escalated root context and `sudo` is used as the `become_method` will reflect the root user:

```
host2.example.com | SUCCESS => {
    "ansible_facts": {
        "ansible_user_dir": "/root",
        "ansible_user_gecos": "root",
        "ansible_user_gid": 0,
        "ansible_user_id": "root",
        "ansible_user_shell": "/bin/bash",
        "ansible_user_uid": 0,
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false
}
``` 

FWIW, if we set the `become_flags` to `-l` for the `su` method we can at least make the facts consistent with what is returned with `sudo`. The user facts seem to be one area reliant on the execution environment. However, it also seems like it could be its own can of worms as to what these facts should default to in the first place - meaning should they follow the escalated user or not. If they should behave like they currently do with `sudo`,  could the default `become_flags` for `su` include `-l`?

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
su / sudo / setup

##### ANSIBLE VERSION

```paste below
ansible 2.9.10
  config file = /Users/ed.velez/.ansible.cfg
  configured module search path = ['/Users/ed.velez/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ed.velez/.pyenv/versions/3.6.10/envs/default/lib/python3.6/site-packages/ansible
  executable location = /Users/ed.velez/.pyenv/versions/default/bin/ansible
  python version = 3.6.10 (default, Jun  3 2020, 15:08:47) [GCC 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.62)]
```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/Users/ed.velez/.ansible.cfg) = True
ANSIBLE_SSH_RETRIES(/Users/ed.velez/.ansible.cfg) = 2
HOST_KEY_CHECKING(/Users/ed.velez/.ansible.cfg) = False
RETRY_FILES_ENABLED(/Users/ed.velez/.ansible.cfg) = False
```

##### OS / ENVIRONMENT

Testing with rhel 7.8

##### STEPS TO REPRODUCE

```yaml
# run something similar to the following on a device with the `su` and `sudo` become methods and compare 
ansible all -m setup -a 'filter=ansible_user_*' -b
```

##### EXPECTED RESULTS

Expected results would be some consistency between `become_methods` for these facts. At the very least it would be great if there was discrepancy on the conditions when they will diverge.

##### ACTUAL RESULTS

See above.

```paste below

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/setup.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/setup.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Using `-l` with `su` isn't in alignment with how we execute `sudo`.  We do not use `-i`/`--login` with `sudo`, and adding that changes the behavior of `sudo` as we would then be loading profile files that we haven't.  So if we added`-l` for `su` that would still be inconsistent with `sudo` unless we added `-i` for sudo.

This is also dependent on where your `su` came from. Whether that is BSD, vs util-linux, vs shadow-utils, vs...

The `util-linux` `su` docs state: 

> For  backward compatibility, su defaults to not change the current directory and to only set the environment
>        variables HOME and SHELL (plus USER and LOGNAME if the target user is not root)

Whereas that from `shadow-utils` does not exhibit that behavior.

Based on the above, I'm leaning toward this being something that the user should define for their environment if it makes sense to do so.
